src/runtime/recipe/handle-connection.ts

```diff
@@ -169,43 +169,42 @@ export class HandleConnection {
   isResolved(options?): boolean {
     assert(Object.isFrozen(this));
 
-    if (this.isOptional) {
-      return true;
-    }
-
     let parent;
     if (this.spec && this.spec.parentConnection) {
       parent = this.particle.connections[this.spec.parentConnection.name];
+      if (!parent || !parent.handle) {
+        if (options) {
+          options.details = 'parent connection missing handle';
+        }
+        return false;
+      }
     }
 
-    // TODO: This should use this._type, or possibly not consider type at all.
-    if (!this.type) {
+    if(!this.handle) {
```

Comment:
I don't think this will pass the automatic lint checks (they enforce a space after the if, I think).

Also I wish we had more lindt checks.

Reply:
Thanks